pandas/core/window/rolling.py

```diff
@@ -374,16 +375,34 @@ def _wrap_result(self, result, block=None, obj=None):
         index = obj.index
 
         if isinstance(result, np.ndarray):
-
             if result.ndim == 1:
                 from pandas import Series
 
                 return Series(result, index, name=obj.name)
+            elif result.ndim == 2 and columns is not None:
+                from pandas import DataFrame
+
+                df = DataFrame(result, index, columns=columns)
+                if isinstance(obj, DataFrame):
+                    pass
+                else:
+                    df.columns.name = obj.name
+                return df
+            elif (
+                result.ndim == 3 and columns is not None
```

Comment:
`result` should always be less than 2 dim

Reply:
Why? This is the whole point on this thing. I can barely remember this change but this is a rolling (time), broadcast (columns), quantile (quantiles). I might have the order wrong in my comment here.